cli/node/node.go

```diff
@@ -0,0 +1,161 @@
+package node
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	crand "crypto/rand"
+
+	"github.com/application-research/filclient/keystore"
+	lmdb "github.com/filecoin-project/go-bs-lmdb"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/lotus/chain/wallet"
+	"github.com/ipfs/go-bitswap"
+	"github.com/ipfs/go-datastore"
+	levelds "github.com/ipfs/go-ds-leveldb"
+	blockstore "github.com/ipfs/go-ipfs-blockstore"
+	"github.com/libp2p/go-libp2p"
+	"github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/metrics"
+	dht "github.com/libp2p/go-libp2p-kad-dht"
+)
+
+type Node struct {
+	Host host.Host
+
+	Datastore  datastore.Batching
+	DHT        *dht.IpfsDHT
+	Blockstore blockstore.Blockstore
+	Bitswap    *bitswap.Bitswap
+
+	Wallet *wallet.LocalWallet
+}
+
+func Setup(ctx context.Context, cfgdir string) (*Node, error) {
+	peerkey, err := loadOrInitPeerKey(keyPath(cfgdir))
+	if err != nil {
+		return nil, err
+	}
+
+	bwc := metrics.NewBandwidthCounter()
+
+	h, err := libp2p.New(
+		libp2p.ListenAddrStrings("/ip4/0.0.0.0/tcp/6766"),
```

Comment:
I suggest we just pick any free port, as we're not expecting anyone to dial this address, it's only used for outgoing connections.
```suggestion
		libp2p.ListenAddrStrings("/ip4/0.0.0.0/tcp/0"),
```